doc/reference/kernel/other/float.rst

```diff
@@ -65,20 +65,38 @@ the kernel supports one or more of the following thread sub-classes:
 * SSE user: A thread that can use both the standard floating point registers
   and SSE registers
 
-The kernel initializes the floating point registers so they can be used
+The kernel initializes and enables access to the floating point registers,
+so they can be used
 by any thread, then saves and restores these registers during
 context switches to ensure the computations performed by each FPU user
 or SSE user are not impacted by the computations performed by the other users.
 
-On the ARM Cortex-M architecture with the Floating Point Extension
-the kernel treats *all* threads
-as FPU users when shared FP registers mode is enabled. This means that the
-floating point registers are saved and restored during each context switch,
-even when the associated threads are not using them. Each thread must provide
-an extra 72 bytes of stack space where these register values can be saved.
-Lazy Stacking is currently not supported in Zephyr applications on ARM
-Cortex-M architecture, and the feature is explicitly disabled during system
-boot.
+On the ARM Cortex-M architecture with the Floating Point Extension the kernel
+treats *all* threads as FPU users when shared FP registers mode is enabled.
+This means that any thread is allowed to access the floating point registers.
+The ARM kernel automatically detects that a given thread is using the floating
+point registers the first time the thread accesses them.
+
+During thread context switching the ARM kernel saves the *callee-saved*
+floating point registers, if the switched-out thread has been using them.
+Additionally, the the *caller-saved* floating point registers are saved on
+the thread's stack. If the switched-in thread has been using the floating
+point registers, the kernel restores the callee-save* FP registers of
+the switched-in thread and the caller-saved FP context is restored from
+the thread's stack. Thus, the kernel does not save or restore the FP
+context of threads that are not using the FP registers.
+
+Each thread that intends to use the floating point registers must provide
+an extra 72 bytes of stack space where the callee-saved FP context can
+be saved.
+
+Lazy Stacking is currently enabled in Zephyr applications on ARM Cortex-M
```

Comment:
Can you provide a reference for "Lazy Stacking" for more information: maybe this if the referenced doc is appropriate:
```
`Lazy Stacking <http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.dai0298a/DAFGGBJD.html>`_
is currently enabled ...
```

Reply:
Yes, I'll add a reference from Arm, thanks

Reply:
Btw, this seems to be a good link :)